internal/controllers/clusterextension_controller.go

```diff
@@ -411,6 +411,11 @@ func (r *ClusterExtensionReconciler) resolve(ctx context.Context, ext ocv1alpha1
 	channelName := ext.Spec.Channel
 	versionRange := ext.Spec.Version
 
+	installedBundle, err := r.installedBundle(ctx, allBundles, &ext)
```

Comment:
This func now in large part resembles `main`. Our feature branch was missing the call to `SuccessorsPredicate` which contains the feature gate switching for semver, which was the primary cause of the e2e failures.